### ISSUE TITLE
Add conditional embedding to Generator and Discriminator classes

### ISSUE THREAD

**Issue by Supergecki** [Supergecki/Scientific-Computing-Project-WiSe2025_26-CIFAR-10-cGAN#6]
Needs #4 and #5 to be resolved.

Embed the class labels using `torch.nn.Embedding` in both `Generator` and `Discriminator` classes and concatenate the class embeddings to the latent vector in the `Generator`.

Maybe embedding dimension 50 would fit?

**Comment by Supergecki** [comment]
@0xBA11E5 Is this resolved as of #33 and #34? If so, we can close it.
